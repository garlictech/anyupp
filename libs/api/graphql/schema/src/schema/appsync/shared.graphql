### COMMON
# TODO: extraxt these because the admin-api also contains these
type Location {
  lat: String!
  lng: String!
}

input LocationInput {
  lat: String!
  lng: String!
}

type Address {
  address: String
  city: String
  country: String
  title: String
  postalCode: String
  location: Location
}

type LocalizedItem {
  en: String
  de: String
  hu: String
}

type PaymentMode {
  name: String!
  caption: String
  method: PaymentMethod!
}
enum PaymentMethod {
  CASH
  CARD
  INAPP
}

type ChainStyle {
  colors: ChainStyleColors!
  images: ChainStyleImages
}

type ChainStyleColors {
  backgroundLight: String!
  backgroundDark: String!
  borderLight: String!
  borderDark: String!
  disabled: String!
  highlight: String!
  indicator: String!
  textLight: String!
  textDark: String!
}

type ChainStyleImages {
  header: String
  logo: String
}
