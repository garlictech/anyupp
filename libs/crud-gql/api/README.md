# crud-gql-api

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test crud-gql-api` to execute the unit tests via [Jest](https://jestjs.io).
