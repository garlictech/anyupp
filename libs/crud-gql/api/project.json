{
  "root": "libs/crud-gql/api",
  "sourceRoot": "libs/crud-gql/api/src",
  "projectType": "library",
  "targets": {
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "options": {
        "lintFilePatterns": ["libs/crud-gql/api/**/*.ts"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/libs/crud-gql/api"],
      "options": {
        "jestConfig": "libs/crud-gql/api/jest.config.ts",
        "passWithNoTests": true
      }
    }
  },
  "tags": ["platform:nodejs", "layer:todo-refactor-old"]
}
