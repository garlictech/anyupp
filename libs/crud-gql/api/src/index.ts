export * from './lib/generated/graphql/mutations';
export * from './lib/generated/graphql/queries';
export * from './lib/generated/graphql/subscriptions';
export * from './lib/generated/api';
export { CrudSdk } from './lib/sdk';
export * from './lib/clients';

import awsmobile from './lib/generated/aws-exports';

export { awsmobile as awsConfig };
