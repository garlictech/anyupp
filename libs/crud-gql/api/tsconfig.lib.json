{
  "extends": "./tsconfig.json",
  "exclude": ["**/*.spec.ts", "**/*.test.ts", "jest.config.ts"],
  "include": ["**/*.ts"],
  "compilerOptions": {
    "noImplicitAny": false
  }
}
