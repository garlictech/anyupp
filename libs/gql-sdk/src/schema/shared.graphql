### Location
type Location @aws_cognito_user_pools {
  lat: Float!
  lng: Float!
}
input LocationInput {
  lat: Float!
  lng: Float!
}

### LocalizedItem
type LocalizedItem @aws_cognito_user_pools {
  en: String
  de: String
  hu: String
}
input LocalizedItemInput {
  en: String
  de: String
  hu: String
}

type Address @aws_cognito_user_pools {
  address: String!
  city: String!
  country: String!
  title: String!
  postalCode: String!
  location: Location!
}

### Payment
type PaymentMode @aws_cognito_user_pools {
  type: PaymentType!
  caption: String
  method: PaymentMethod!
}

enum PaymentMethod {
  cash
  card
  inapp
}

enum PaymentType {
  card
  cash
  simple
  stripe
  applepay
  googlepay
}

### ChainStyle
type ChainStyle @aws_cognito_user_pools {
  colors: ChainStyleColors!
  images: ChainStyleImages
}

type ChainStyleColors @aws_cognito_user_pools {
  backgroundLight: String!
  backgroundDark: String!
  borderLight: String!
  borderDark: String!
  disabled: String!
  highlight: String!
  indicator: String!
  textLight: String!
  textDark: String!
}

type ChainStyleImages @aws_cognito_user_pools {
  header: String
  logo: String
}

### ChainProduct
type ChainProduct
  @aws_cognito_user_pools
  @model
  @searchable
  @auth(rules: [{ allow: private }, { allow: public, provider: iam }]) {
  id: ID!
  chainId: ID!
  name: LocalizedItem!
  description: LocalizedItem
  productCategoryId: ID!
  productType: String!
  isVisible: Boolean!
  image: String
  variants: [ProductVariant]
  allergens: [Allergen]
  configSets: [ProductConfigSet]
}
input UpdateChainProductInput {
  id: ID!
  chainId: ID
  name: LocalizedItemInput
  description: LocalizedItemInput
  productCategoryId: ID
  productType: String
  isVisible: Boolean
  image: String
  variants: [ProductVariantInput]
  allergens: [Allergen]
  configSets: [ProductConfigSetInput]
}

enum Allergen {
  celery
  crustaceans
  egg
  fish
  gluten
  lupin
  milk
  molluscs
  mustard
  peanut
  sesame
  soya
  sulphites
  treenuts
}

enum ServingMode {
  inPlace
  takeAway
}

enum OrderMode {
  instant
  pickup
}

### GroupProduct
type GroupProduct
  @aws_cognito_user_pools
  @model
  @searchable
  @auth(rules: [{ allow: private }, { allow: public, provider: iam }]) {
  id: ID!
  parentId: ID! # parent chainProduct ID
  chainId: ID!
  groupId: ID!
  isVisible: Boolean!
  tax: Int!
  takeawayTax: Int
  variants: [ProductVariant]
  configSets: [ProductConfigSet]
}
input UpdateGroupProductInput {
  id: ID!
  parentId: ID
  chainId: ID
  groupId: ID
  isVisible: Boolean
  tax: Int
  takeawayTax: Int
  variants: [ProductVariantInput]
  configSets: [ProductConfigSetInput]
}

### UnitProduct
type UnitProduct
  @aws_cognito_user_pools
  @model
  @searchable
  @auth(rules: [{ allow: private }, { allow: public, provider: iam }]) {
  id: ID!
  parentId: ID! # parent groupProduct ID
  chainId: ID!
  groupId: ID!
  unitId: ID!
  isVisible: Boolean!
  takeaway: Boolean! @deprecated(reason: "Use `supportedServingModes`.")
  laneId: ID
  position: Int!
  variants: [ProductVariant]
  configSets: [ProductConfigSet]
  supportedServingModes: [ServingMode!]!
}
input CreateUnitProductInput {
  id: ID # <== NOT REQUIRED
  parentId: ID! # parent groupProduct ID
  chainId: ID!
  groupId: ID!
  unitId: ID!
  isVisible: Boolean!
  takeaway: Boolean!
  laneId: ID
  position: Int!
  variants: [ProductVariantInput]
  configSets: [ProductConfigSetInput]
  supportedServingModes: [ServingMode!]!
}
input UpdateUnitProductInput {
  id: ID! # <== REQUIRED
  parentId: ID
  chainId: ID
  groupId: ID
  unitId: ID
  isVisible: Boolean
  takeaway: Boolean
  laneId: ID
  position: Int
  variants: [ProductVariantInput]
  configSets: [ProductConfigSetInput]
  supportedServingModes: [ServingMode!]
}

### ProductConfigSet
type ProductConfigComponent @aws_cognito_user_pools {
  productComponentId: ID!
  refGroupPrice: Float!
  price: Float!
  position: Int!
}
input ProductConfigComponentInput {
  productComponentId: ID!
  refGroupPrice: Float!
  price: Float!
  position: Int!
}

type ProductConfigSet @aws_cognito_user_pools {
  productSetId: ID!
  items: [ProductConfigComponent!]!
  position: Int!
}
input ProductConfigSetInput {
  productSetId: ID!
  items: [ProductConfigComponentInput!]!
  position: Int!
}

### ProductVariant
type ProductVariant @aws_cognito_user_pools {
  id: ID!
  variantName: LocalizedItem!
  pack: ProductVariantPack
  refGroupPrice: Float
  isAvailable: Boolean!
  price: Float!
  availabilities: [Availability]
  position: Int!
}
input ProductVariantInput {
  id: ID!
  variantName: LocalizedItemInput!
  pack: ProductVariantPackInput
  refGroupPrice: Float
  isAvailable: Boolean!
  price: Float!
  availabilities: [AvailabilityInput]
  position: Int!
}

### ProductVariantPack
type ProductVariantPack @aws_cognito_user_pools {
  size: Float!
  unit: String!
}
input ProductVariantPackInput {
  size: Float!
  unit: String!
}

### Availability
type Availability @aws_cognito_user_pools {
  type: String!
  dayFrom: String
  dayTo: String
  timeFrom: String
  timeTo: String
  price: Float!
}
input AvailabilityInput {
  type: String!
  dayFrom: String
  dayTo: String
  timeFrom: String
  timeTo: String
  price: Float!
}

### UserInvoiceAddress
# Re enable (issue: #1736)
# type UserInvoiceAddress {
#   customerName: String!
#   taxNumber: String!
#   country: String!
#   city: String!
#   streetAddress: String!
#   postalCode: String!
#   email: String
# }
# input UserInvoiceAddressInput {
#   customerName: String!
#   taxNumber: String!
#   country: String!
#   city: String!
#   streetAddress: String!
#   postalCode: String!
#   email: String
# }
