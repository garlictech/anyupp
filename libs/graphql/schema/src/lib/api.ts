
/** ------------------------------------------------------
 * THIS FILE WAS AUTOMATICALLY GENERATED (DO NOT MODIFY)
 * -------------------------------------------------------
 */

/* tslint:disable */
/* eslint-disable */
export class UpdateAdminUserInput {
    email?: string;
    name?: string;
    phone?: string;
}

export class AdminUser {
    __typename?: 'AdminUser';
    email?: string;
    name?: string;
    phone?: string;
}

export abstract class IQuery {
    __typename?: 'IQuery';

    abstract hello(): string | Promise<string>;

    abstract getAdminUser(id: string): AdminUser | Promise<AdminUser>;
}

export abstract class ISubscription {
    __typename?: 'ISubscription';

    abstract adminUserChanged(id: string): AdminUser | Promise<AdminUser>;
}

export abstract class IMutation {
    __typename?: 'IMutation';

    abstract updateAdminUser(id: string, newAdminData: UpdateAdminUserInput): boolean | Promise<boolean>;
}
