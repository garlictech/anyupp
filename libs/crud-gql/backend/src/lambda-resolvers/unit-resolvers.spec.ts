import * as CrudApi from '@bgap/crud-gql/api';
import {
  UnitResolverDeps,
  createUnitResolver,
  updateUnitResolver,
} from './unit-resolvers';
import { DynamoDB } from 'aws-sdk';
import { unitFixture } from '@bgap/shared/fixtures';

describe('Test custom CRUD resolvers', () => {
  const getDeps = (): UnitResolverDeps => ({
    hashGenerator: (param: string) => Promise.resolve(`HASHED ${param}`),
    tableName: 'UNIT_TABLE_NAME',
    docClient: new DynamoDB.DocumentClient(),
    uuidGenerator: () => 'THE_UUID',
  });

  const getFailedDeps = (): UnitResolverDeps => ({
    hashGenerator: (_param: string) => Promise.reject(`ERROR`),
    tableName: 'UNIT_TABLE_NAME',
    docClient: new DynamoDB.DocumentClient(),
    uuidGenerator: () => 'THE_UUID',
  });

  const fixtureNoPassword: CrudApi.CreateUnitInput = {
    ...unitFixture.unitBase,
    chainId: 'CHAINID',
    groupId: 'GROUPID',
  };

  const fixtureWithPassword: CrudApi.CreateUnitInput = {
    ...unitFixture.unitBase,
    chainId: 'CHAINID',
    groupId: 'GROUPID',
    pos: {
      type: CrudApi.PosType.rkeeper,
      rkeeper: {
        endpointUri: 'ENDPOINT',
        rkeeperUsername: 'USERNAME',
        rkeeperPassword: 'PASSWORD',
        anyuppUsername: 'ANYUPPUSERNAME',
        anyuppPassword: 'PASSWORD2',
        restaurantId: 'RESTAURANTID',
      },
    },
  };

  const cases = [fixtureNoPassword, fixtureWithPassword];

  test.each(cases)('createUnitResolver tests', async fixture => {
    const deps: UnitResolverDeps = getDeps();

    const putSpy = jest.spyOn(deps.docClient, 'put');
    const updateSpy = jest.spyOn(deps.docClient, 'update');

    expect(await createUnitResolver(deps)(fixture).toPromise()).toEqual(true);
    expect(putSpy.mock.calls).toMatchSnapshot('PUT CALL');
    expect(updateSpy).not.toHaveBeenCalled();

    const failedDeps = getFailedDeps();
    expect(createUnitResolver(failedDeps)(fixture).toPromise()).rejects.toEqual(
      'ERROR',
    );
  });

  test.each(cases)('updateUnitResolver tests', async fixture => {
    const deps: UnitResolverDeps = getDeps();
    const putSpy = jest.spyOn(deps.docClient, 'put');
    const updateSpy = jest.spyOn(deps.docClient, 'update');

    expect(
      await updateUnitResolver(deps)({ ...fixture, id: 'ID' }).toPromise(),
    ).toEqual(true);
    expect(updateSpy.mock.calls).toMatchSnapshot('PUT CALL');
    expect(putSpy).not.toHaveBeenCalled();

    const failedDeps = getFailedDeps();
    expect(createUnitResolver(failedDeps)(fixture).toPromise()).rejects.toEqual(
      'ERROR',
    );
  });
});
