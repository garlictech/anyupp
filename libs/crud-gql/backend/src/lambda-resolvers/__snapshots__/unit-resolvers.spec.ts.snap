// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test custom CRUD resolvers createUnitResolver tests: PUT CALL 1`] = `
Array [
  Array [
    Object {
      "Item": Object {
        "address": Object {
          "address": "Ág u. 1.",
          "city": "Budapest",
          "country": "Magyarország",
          "location": Object {
            "lat": 47,
            "lng": 19,
          },
          "postalCode": "1021",
          "title": "HQ",
        },
        "chainId": "CHAINID",
        "description": Object {
          "en": "Test unit #test_unit_1_id description",
          "hu": "Teszt unit #test_unit_1_id leírás",
        },
        "groupId": "GROUPID",
        "id": "THE_UUID",
        "isAcceptingOrders": true,
        "isActive": true,
        "lanes": Array [
          Object {
            "color": "#e72222",
            "id": "lane_01",
            "name": "bár",
          },
          Object {
            "color": "#e123ef",
            "id": "lane_02",
            "name": "konyha",
          },
        ],
        "name": "Késdobáló S",
        "open": Object {
          "from": "1970-01-01",
        },
        "paymentModes": Array [
          Object {
            "method": "cash",
            "type": "cash",
          },
          Object {
            "method": "card",
            "type": "card",
          },
          Object {
            "method": "inapp",
            "type": "stripe",
          },
        ],
      },
      "TableName": "Unit-ttk4poqlpjbudlhzfpvbz2ff7q-staging",
    },
  ],
]
`;

exports[`Test custom CRUD resolvers createUnitResolver tests: PUT CALL 2`] = `
Array [
  Array [
    Object {
      "Item": Object {
        "address": Object {
          "address": "Ág u. 1.",
          "city": "Budapest",
          "country": "Magyarország",
          "location": Object {
            "lat": 47,
            "lng": 19,
          },
          "postalCode": "1021",
          "title": "HQ",
        },
        "chainId": "CHAINID",
        "description": Object {
          "en": "Test unit #test_unit_1_id description",
          "hu": "Teszt unit #test_unit_1_id leírás",
        },
        "groupId": "GROUPID",
        "id": "THE_UUID",
        "isAcceptingOrders": true,
        "isActive": true,
        "lanes": Array [
          Object {
            "color": "#e72222",
            "id": "lane_01",
            "name": "bár",
          },
          Object {
            "color": "#e123ef",
            "id": "lane_02",
            "name": "konyha",
          },
        ],
        "name": "Késdobáló S",
        "open": Object {
          "from": "1970-01-01",
        },
        "paymentModes": Array [
          Object {
            "method": "cash",
            "type": "cash",
          },
          Object {
            "method": "card",
            "type": "card",
          },
          Object {
            "method": "inapp",
            "type": "stripe",
          },
        ],
        "pos": Object {
          "rkeeper": Object {
            "anyuppPassword": "HASHED PASSWORD2",
            "anyuppUsername": "ANYUPPUSERNAME",
            "endpointUri": "ENDPOINT",
            "restaurantId": "RESTAURANTID",
            "rkeeperPassword": "HASHED PASSWORD",
            "rkeeperUsername": "USERNAME",
          },
          "type": "rkeeper",
        },
      },
      "TableName": "Unit-ttk4poqlpjbudlhzfpvbz2ff7q-staging",
    },
  ],
]
`;
